src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -176,164 +127,171 @@ const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
     bgcolor: 'background.paper',
   };
 
-  // return null if we don't have the workflow loaded.
-  // This workflow doesn't exist.
-  if (workflow.loadingStatus.loading === LoadingStatusEnum.Failed) {
-    navigate('/404');
-    return null;
-  }
-
-  if (operator) {
-    const inputs = operator.inputs;
-    const outputs = operator.outputs;
-
-    const operatorArtifacts = [...inputs, ...outputs];
-    // fetch output artifacts.
-    operatorArtifacts.map((artifactId) => {
-      if (!workflow.artifactResults[artifactId])
-        dispatch(
-          handleGetArtifactResults({
-            apiKey: user.apiKey,
-            workflowDagResultId,
-            artifactId,
-          })
-        );
-    });
+  if (
+    !workflowDagResultWithLoadingStatus ||
+    isInitial(workflowDagResultWithLoadingStatus.status) ||
+    isLoading(workflowDagResultWithLoadingStatus.status)
+  ) {
+    return (
+      <Layout user={user}>
+        <CircularProgress />
+      </Layout>
+    );
   }
 
-  if (!metricResult || !metricResult.result) {
+  if (isFailed(workflowDagResultWithLoadingStatus.status)) {
     return (
       <Layout user={user}>
-        <CircularProgress />
+        <Alert title="Failed to load workflow">
+          {workflowDagResultWithLoadingStatus.status.err}
+        </Alert>
       </Layout>
     );
   }
 
   // Function to get the numerical value of the metric output
-  const getOperatorOutput = () => {
-    if (!operator || !operator.outputs) {
-      return <CircularProgress />;
+  const operatorOutputsList = operator.outputs.map((artifactId) => {
+    const artifactResult = (workflowDagResultWithLoadingStatus.result
+      ?.artifacts ?? {})[artifactId];
+    if (!artifactResult) {
+      return null;
     }
 
-    return operator.outputs.map((artifactId) => {
-      const artifactResult = workflow.artifactResults[artifactId];
-      if (!artifactResult) {
-        return null;
-      }
-
-      // TODO: Check the serialization_type of the artifacts and show a link
-      // to table vew artifacts when present.
-      // TODO: Do the same thing that we're doing over in inputs :)
-
-      if (artifactResult.result) {
-        if (artifactResult.result.artifact_type === 'table') {
-          // Link to appropriate artifact details page
-          // Show tableIcon here as part of the link.
-          return (
-            <Box key={artifactId}>
-              <Typography variant="body1">TEST</Typography>
-            </Box>
-          );
-        } else {
-          // Render inline if possible
-          return (
-            <Box>
-              <Typography variant="body1">
-                {artifactResult.result.data}
-              </Typography>
-            </Box>
-          );
-        }
-      }
+    // TODO: Check the serialization_type of the artifacts and show a link
```

Comment:
I'm assuming we plan on dealing with these TODOs in a later PR? Maybe we should create tasks for these in Linear